src/devhub/entity/community/Discussions.jsx

```diff
@@ -0,0 +1,247 @@
+);
+
+getCommunity = getCommunity || (() => <></>);
+setCommunitySocialDB = setCommunitySocialDB || (() => <></>);
+
+const communityData = getCommunity({ handle });
+const [postsExists, setPostExists] = useState(false);
+
+const MainContent = styled.div`
+  padding-left: 2rem;
+  flex: 3;
+  @media screen and (max-width: 960px) {
+    padding-left: 0rem;
+  }
+  .post:hover {
+    background-color: inherit !important;
+  }
+`;
+
+const SidebarContainer = styled.div`
+  flex: 1;
+`;
+
+const Heading = styled.div`
+  font-size: 19px;
+  font-weight: 600;
+`;
+
+const SubHeading = styled.div`
+  font-size: 15px;
+  font-weight: 600;
+`;
+
+const Container = styled.div`
+  flex-wrap: no-wrap;
+  max-width: 100%;
+
+  .max-width-100 {
+    max-width: 100%;
+  }
+  @media screen and (max-width: 960px) {
+    flex-wrap: wrap;
+  }
+
+  .card {
+    border-radius: 1rem !important;
+  }
+
+  .display-none {
+    display: none;
+  }
+`;
+
+const Tag = styled.div`
+  border-top-right-radius: 50px;
+  border-bottom-right-radius: 50px;
+  border-top-left-radius: 50px;
+  border-bottom-left-radius: 50px;
+  padding-inline: 0.8rem;
+  padding-block: 0.3rem;
+  display: flex;
+  gap: 0.5rem;
+  border-width: 1px;
+  border-style: solid;
+  font-size: 14px;
+  color: rgba(0, 236, 151, 1);
+  font-weight: 800;
+`;
+
+const [sort, setSort] = useState("timedesc");
+
+function repostOnDiscussions(blockHeight) {
+  Near.call([
+    {
+      contractName: "${REPL_DEVHUB_CONTRACT}",
+      methodName: "create_discussion",
+      args: {
+        handle,
+        block_height: blockHeight,
+      },
+      gas: Big(10).pow(14),
+    },
+  ]);
+}
+
+async function checkHashes() {
+  if (props.transactionHashes) {
+    asyncFetch("https://rpc.mainnet.near.org", {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "dontcare",
+        method: "tx",
+        params: [props.transactionHashes, context.accountId],
+      }),
+    }).then((transaction) => {
+      if (transaction !== null) {
+        const transaction_method_name =
+          transaction?.body?.result?.transaction?.actions[0].FunctionCall
+            .method_name;
+
+        if (transaction_method_name === "set") {
+          getBlockHeightAndRepost();
+        }
+
+        // show the latest created post to user
+        if (transaction_method_name === "create_discussion") {
+          console.log("Discussions created in the last call, show it to user.");
+        }
+      }
+    });
+  }
+}
+
+function getBlockHeightAndRepost() {
+  Near.asyncView("${REPL_SOCIAL_CONTRACT}", "get", {
+    keys: [`${context.accountId}/post/**`],
+    options: {
+      with_block_height: true,
+    },
+  })
+    .then((response) => {
+      let blockHeight = response[context.accountId][":block"];
+      repostOnDiscussions(blockHeight);
+    })
+    .catch((error) => {
+      console.log("error", error);
```

Comment:
Let's be more descriptive in the log messages, so it is easier to at least find the relevant code when you see the error in the console logs. For example:

```suggestion
      console.log("DevHub Error [Discussions]: getBlockHeightAndRepost failed", error);
```